aclk/aclk_lws_wss_client.c

```diff
@@ -427,6 +461,8 @@ static int aclk_lws_wss_callback(struct lws *wsi, enum lws_callback_reasons reas
             return -1;                       // the callback response is ignored, hope the above remains true
         case LWS_CALLBACK_WSI_DESTROY:
             aclk_lws_wss_clear_io_buffers(engine_instance);
+            if (!engine_instance->websocket_connection_up)
```

Comment:
This is not always called. When I messed the password (after the challenge) the connection was not established.

Why does it expect to have this `websocket_connection_up == 0` when the code that sets this to 0 is L467

Reply:
We want report to be sent only if WSI_DESTROY is called when there was not successful connection yet.
Idea is to report when ACLK link could not be established.

Reply:
Point of this is to have an idea in how many cases ACLK cannot connect (e.g. by blocked firewall).

Reply:
we are diagnosing if connection to the cloud was possible (not blocked by firewall) as soon as connection is established we know that is OK and we don't have to report anything.
It doesn't try to diagnose the MQTT failures that might happen after.

Reply:
Why not use the state of `aclk_connected`? This is the high-level flag that the link was established correctly. Your logic that any WSI_DESTROY before success is a form of failure would apply better to this flag.

Reply:
that is too late... we do not care about MQTT here. If connection TLS upgrade and then upgrade to Websockets succeeds then connection succeeded (proxy or firewall didn't block us)

Reply:
Yes we do - that is the whole point.